MatrixSDK/ClientInformation/MXClientInformationService.swift

```diff
@@ -0,0 +1,90 @@
+// 
+// Copyright 2022 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+@objcMembers
+public class MXClientInformationService: NSObject {
+
+    private weak var session: MXSession?
```

Comment:
Any reason `session` is `weak`?

Reply:
To avoid retain cycles between session and the service because the session holds a strong reference to the service.

Reply:
Oh, right. The session holds the reference 👍